src/modules/sync/utils/contentTypeHelpers.ts

```diff
@@ -1,41 +1,41 @@
 import { ContentTypeElements } from "@kontent-ai/management-sdk";
-import {
-  DomHtmlNode,
-  DomNode,
-  nodeParse,
-  ParseResult,
-  ResolveDomHtmlNodeType,
-  ResolveDomTextNodeType,
-  transformToJson,
-} from "@kontent-ai/rich-text-resolver";
+import { PortableTextObject, transformToPortableText } from "@kontent-ai/rich-text-resolver";
 
 import { notNullOrUndefined } from "../../../utils/typeguards.js";
 import { getAssetReferences, getItemReferences, OriginalReference } from "../diff/guidelinesRichText.js";
 import { SyncAssetElement, SyncLinkedItemsElement, SyncTypeElement } from "../types/syncModel.js";
 
-const resolveAssetIdsDomHtmlNode: ResolveDomHtmlNodeType = (node, traverse) => {
-  switch (node.tagName) {
-    case "figure":
-      return node.attributes["data-asset-id"];
-    case "a":
-      return node.attributes["data-asset-id"];
-    default: {
-      return node.children.map(traverse);
-    }
+const resolveAssetIdsDomHtmlNode = (node: PortableTextObject): ReadonlyArray<string> => {
+  if (node._type === "image") {
+    return node.asset.referenceType === "id" ? [node.asset._ref] : [];
+  }
+
+  if (node._type === "block") {
+    return node.markDefs?.filter(a => a._type === "link" && "data-asset-id" in a).map(a => (a as any)["data-asset-id"])
+      ?? [];
   }
-};
 
-const resolveItemIdsDomHtmlNode: ResolveDomHtmlNodeType = (node, traverse) => {
-  switch (node.tagName) {
-    case "a":
-      return node.attributes["data-item-id"];
-    default: {
-      return node.children.flatMap(traverse);
-    }
+  if (node.children && Array.isArray(node.children)) {
+    return node.children.flatMap(resolveAssetIdsDomHtmlNode);
   }
+
+  return [];
 };
 
-const customResolveDomTextNode: ResolveDomTextNodeType = () => null;
+const resolveItemIdsDomHtmlNode = (node: PortableTextObject): ReadonlyArray<string> => {
+  if (node._type === "componentOrItem") {
+    return node.dataType === "item" ? [node.componentOrItem._ref] : [];
+  }
+  if (node._type === "block") {
+    return node.markDefs?.filter(a => a._type === "contentItemLink").map(a => (a as any).contentItemLink._ref) ?? [];
```

Comment:
Can you use this type instead of any? 
<img width="683" alt="image" src="https://github.com/user-attachments/assets/2c9c34ea-4416-46a7-8035-4024c326fab0" />
Also, according to the type the `a.contentItemLink._ref` is an object, not a string. Does the code work?

Reply:
no contentItemLink is reference ._ref is a string :D however casting to this object sounds better, thanks :)

also the `getRequiredIds` that utilizes this functions are tested, so yeah it should be working :)

Reply:
Oh, you are right